src/uct/ib/base/ib_device.c

```diff
@@ -482,6 +482,66 @@ void uct_ib_handle_async_event(uct_ib_device_t *dev, uct_ib_async_event_t *event
     ucs_log(level, "IB Async event on %s: %s", uct_ib_device_name(dev), event_info);
 }
 
+static ucs_status_t uct_ib_device_get_path_buffer(uct_ib_device_t *dev,
+                                                  char *path_buffer)
+{
+    char *resolved_path;
+
+    resolved_path = realpath(dev->ibv_context->device->ibdev_path, path_buffer);
+    if (resolved_path == NULL) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    /* Make sure there is "/infiniband/" substring in path_buffer */
+    if (strstr(path_buffer, "/infiniband/") == NULL) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    return UCS_OK;
+}
+
+static ucs_status_t uct_ib_device_get_sf_ids(uct_ib_device_t *dev)
+{
+    long vendor_id, device_id;
+    char path_buffer[PATH_MAX];
+    char pci_bus[UCT_IB_PCI_BUS_ID_LEN + 1];
+    char *bus_id;
+    ucs_status_t status;
+
+    /* realpath name is of form /sys/devices/.../0000:03:00.0/<UUID>/infiniband/mlx5_0
+     * and bus_id is constructed from 0000:03:00.0 */
+
+    status = uct_ib_device_get_path_buffer(dev, path_buffer);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    bus_id = basename(dirname(dirname(dirname(path_buffer))));
+
+    /* get 0000:03 from 0000:03:00.0*/
+    memcpy(pci_bus, bus_id, UCT_IB_PCI_BUS_ID_LEN);
+    pci_bus[UCT_IB_PCI_BUS_ID_LEN] = 0;
```

Comment:
can we unite common code with uct_ib_device_get_sys_dev ?

Reply:
@yosefe This is not common code with uct_ib_device_get_sys_dev. uct_ib_device_get_sys_dev doesn't need to use 0000:03. It only needs 0000:03:00.0.